### DISCUSSION TITLE
Cannot push multiple files to the repository.

### DISCUSSION THREAD

**Discussion by ozankrds** [community/community#134177] [Repositories] [answered]
### Select Topic Area

Question

### Body

I made a qr code generator using node.js so there are a lot of files that I don't know about. I'm trying to push them using github desktop and also bash, but I always get an error of "! [remote rejected] main -> main (push declined due to repository rule violations)". I did set my email public (That's a solution I've seen here, but it didn't work). What should I do?
![Ekran görüntüsü 2024-07-31 163410](https://github.com/user-attachments/assets/f0a0e09a-b276-4d1f-b151-9f9695eb0fbd)

In the link below secret, there is this which I think might have caused the error
![Ekran görüntüsü 2024-07-31 163746](https://github.com/user-attachments/assets/cfdbc7f5-6aab-4750-b1ff-0cc55687e179)

**Comment by patrick-knight** [comment, marked as answer]
Secret scanning is alerting you to the fact `node_modules/public-encrypt/test/test_rsa_privkey-pem` has an SSH key present in it, which is against best practices to store in a repository. You best course of action would to remove that file from your commit and try pushing again. 

If you push that SSH key you should assume it is likely compromised and should be rotated.

**Reply by 7GenerationTech** [reply]
What would be the point of it being compromised?

**Reply by ozankrds** [reply]
Thank you. It helped me.
